liquibase-standard/src/main/java/liquibase/change/AbstractChange.java

```diff
@@ -818,4 +837,29 @@ public String getDescription() {
 
         return description;
     }
+
+    /**
+     * This method will read the content of the given stream and make any parameter update on the content. For example,
+     * making a property replacement into procedure text read from the sql file.
+     * @param stream
+     * @return an updated {@link InputStream} if any replacement has been performed in the content of the original stream.
+     */
+    protected InputStream updateStreamContentIfApplicable(InputStream stream) {
```

Comment:
I like the idea of this being a stream wrapper, but the function name is not very obvious what its for.  In thinking about a better name, I'm thinking it's maybe best to have it be a stand-alone InputStream class, like normally found? That makes it more usable for other places that may want to expand expressions on a stream beyond just a Change.

So maybe make a new `liquibase.changelog.PropertyExpandingStream` class as a wrapper around the updatedContentStream ByteArrayInputStream created in the function? Instead, make that updatedContentStream in the constructor and then have the various stream read methods call down to the updatedContentStream?

In theory it would be nice to not read the whole file into memory to do that, but that can be a later performance improvement if/when it's a problem.

Reply:
Done